### PR TITLE
Add tutorial image segmentation sample notebook / Streamline Tutorial Onboarding

### LINKED ISSUE #383: [Enhancement/Bug] Streamline Tutorial Onboarding and Resolve Silent Inference Failures
The current tutorial_image_segmentation.ipynb has a high barrier to entry and fails silently on specific local environments.

**1.     Massive Data Overhead:** The requirement to download 10GB of data just to verify the pipeline works is a major blockage. It often triggers Google Drive download limits and discourages quick testing.

**2.    Silent Inference Failure:** On certain Intel-based hardware (MacBook Pro 2016), the model runs without errors but produces an all-zero prediction mask. I checked whether the weights are loaded normally (they seem fine) and the range of input values (0-255).

	Prediction  is 0s 

Image

         Weights are OK	

Image

       Input values are OK

Image

**3.    Environment Incompatibility:** The notebook fails in Google Colab due to a version mismatch with numpy, Colab comes with numpy > 2.2.x while perceptionmetrics requires 1.26.2.

**SOLUTIONS:**
I propose updating the tutorial with the following improvements:

**1. "Quick Start" Mode (Sample Dataset)**
Introduce a v_small dataset option. I would introduce a curated set of 5–10 images instead of the full 10GB. Users can verify their environment and the inference logic in under 60 seconds.
**3. Robust Environment Support (Colab & NumPy)**
Here I am still working on finding the right combination. Colab's environment update to NumPy 2.x breaks backward compatibility with the current requirement of 1.26.2. 
Plus, here I would add instructions on how to create a mount such that users don't need to download the data every time.

### PR DESCRIPTION
**Pull request by iocalangiu**
Hi,

I created a new notebook that downloads a small sample of the RELLIS-3D dataset. The structure of the notebook is identical to tutorial_image_segmentation.ipynb, except now the links point to my google drive where the sample is stored. I also modified the header at the top of the file to clarify the difference from the original notebook.

image

If no other changes are required, could you please re-run the cells after acceptance? For me the inference part produces a black output (same problem I have also with the original notebook - something with my laptop). Would be good to have it well rendered online.

### CODE REVIEW

**Comment by dpascualhe** [comment]
Hi @iocalangiu , thanks for your contribution! :smile: 

The notebook is working perfectly fine for me. Can you check that all the dependencies in your environment are up to date? Maybe you can play around with the matplotlib based visualization as it might be problematic in Google Colab.

I've uploaded the data that you have sampled to the org drive so that you don't have to host yourself: https://drive.google.com/drive/folders/1L-G8fWTwrXnfzOmIdf3gEZd3h32V-HqB?usp=sharing . Can you update the IDs for downloading the data?

We can merge after that!

**Comment by iocalangiu** [comment]
Actually the prediction worked on Google Colab. It was just a problem on my laptop, but found the cause.

The following code sets  ```self.device = 'mps'``` for me, even though my Mac doesn't have mps.

torch_segmentation.py:230-235:

```python
self.device = torch.device(
    "cuda" if torch.cuda.is_available()
    else "mps" if torch.backends.mps.is_available()
    else "cpu"
)
```

I checked in the notebook:
```python
import torch
print("mps available:", torch.backends.mps.is_available())
print("mps built:", torch.backends.mps.is_built())
```
mps available: True
mps built: True

Changing locally in the notebook:
```python
import torch
torch.backends.mps.is_available = lambda: False
```

Generated a correct prediction:
image

Let me know if I should open a new Issue/PR to fix it?

**Comment by iocalangiu** [comment]
Hi @dpascualhe, just wanted to check what is the status about updating the sample notebook.

**Comment by dpascualhe** [comment]
Hi @iocalangiu ! Sorry for the delay. I think we can merge now and keep an eye out in case someone else has this issue with MPS backend. Thanks a lot for your contribution!
